### PR TITLE
Update `AlignableTracker` for phase-2 Outer Tracker

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The goal of this PR is to introduce the necessary changes in order to be able to construct the `AlignableTracker` geometry tree in the case of the phase-2 Tracker detector.
For that purpose I introduce a new class `AlignableStackDet` to deal with the Stack detectors (PS and SS modules) of the Phase-2 Outer Tracker and use it in the new method `AlignableTrackerBuilder::buildOuterTrackerDetector`.
I also add a test file to create random misalignment scenarios starting from the D88 (T21) and a couple of misalignment scenarios for Phase2. 

**This PR is still draft, pending some more refined validation from the alignment group**. 
I would nevertheless profit of the central testing infrastructure to make sure this doesn't create issues in the existing workflows.

#### PR validation:

Relies on the augmented unit tests introduced in this PR:

```
scram b runtests_test_CreateRandomMisalignment
```

runs fine.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

For the moment no backport is foreseen, might happen at a later time.

cc:
@consuegs @henriettepetersen

### CODE REVIEW

File: Alignment/TrackerAlignment/src/AlignableStackDet.cc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  if (units.size() != 2 || stackedDet->lowerDet()->geographicalId() != units[0]->geomDetId() ||
+      stackedDet->upperDet()->geographicalId() != units[1]->geomDetId()) {
+    throw cms::Exception("LogicError") << "[AlignableStackDet] "
+                                       << "Either != 2 components or "
+                                       << "upper/lower in wrong order for consistifyAlignments.";
+  }
+}
+
+//__________________________________________________________________________________________________
+AlignableStackDet::~AlignableStackDet() {}
```
Comment (tvami) [inline Alignment/TrackerAlignment/src/AlignableStackDet.cc:33]:
```suggestion
AlignableStackDet::~AlignableStackDet() = default;
```

**Comment by mmusich** [comment]
test parameters:
  * addpkg = Alignment/MillePedeAlignmentAlgorithm

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/TrackerAlignment/src/AlignableStackDet.cc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      stackedDet->upperDet()->geographicalId() != units[1]->geomDetId()) {
+    throw cms::Exception("LogicError") << "[AlignableStackDet] "
+                                       << "Either != 2 components or "
+                                       << "upper/lower in wrong order for consistifyAlignments.";
+  }
+}
+
+//__________________________________________________________________________________________________
+Alignments* AlignableStackDet::alignments() const {
+  const_cast<AlignableStackDet*>(this)->consistifyAlignments();
```
[the remark this answers is not in the document]
Comment (mmusich) [reply Alignment/TrackerAlignment/src/AlignableStackDet.cc:34]: right.
The same pattern is already present in another class of the same package https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e2b5a1/31523/llvm-analysis/report-93dab5.html#EndPath and getting rid of it entails more complicated changes that are outside the scope of this PR. On the bright side, there shouldn't be further proliferation after this PR as this exhaust the types of Tracker `GeomDet`s to be treated.

[the remark this answers is not in the document]
Comment (tvami) [reply Alignment/TrackerAlignment/src/AlignableStackDet.cc:34]: ok thanks

**Comment by mmusich** [comment]
to update on this thread as well.
@consuegs has run some validation using the misalignment scenarios produced using this PR, results are available [here](https://sconsueg.web.cern.ch/sconsueg/Alignment/Validation_MCscenarioPhase2/PV_validation/). 
I will release this for review once the full set becomes available.

**Comment by mmusich** [comment]
Opening for review based on the results collected [here](https://indico.cern.ch/event/1271467/#25-validation-of-phase-2-misal)

**Comment by tvami** [comment]
+alca

**Comment by rappoccio** [comment]
+1

**Pull request merged**